src/components/search/GlobalSearchField.js

```diff
@@ -3,139 +3,485 @@
+        margin: 0,
+    },
+    option: {
+        minHeight: "auto",
+        padding: theme.spacing(0),
+        margin: theme.spacing(0),
     },
     searchFilter: {
         marginRight: theme.spacing(4),
+        height: theme.spacing(3.6),
+        borderRadius: 0,
+        width: 100,
+        color: theme.palette.info.main,
         [theme.breakpoints.down("xs")]: {
             margin: theme.spacing(1),
-            width: 90,
+            borderRadius: theme.spacing(1),
+            width: "90%",
+        },
+        [theme.breakpoints.up("sm")]: {
+            backgroundColor: theme.palette.info.contrastText,
+        },
+    },
+    input: {
+        position: "relative",
+        height: theme.spacing(3.6),
+        borderRadius: 0,
+        color: theme.palette.info.main,
+        [theme.breakpoints.down("xs")]: {
+            border: theme.spacing(0.5),
+            borderRadius: theme.spacing(1),
+        },
+        [theme.breakpoints.up("sm")]: {
+            backgroundColor: theme.palette.info.contrastText,
         },
     },
+    icon: {
+        marginTop: theme.spacing(2),
+        marginBotton: 0,
+        marginLeft: 0,
+        marginRight: 0,
+    },
 }));
 
+//gobal search constants
+const PAGE = 0;
+const ROWS = 10;
+
+const SearchOption = React.forwardRef((props, ref) => {
+    const { primary, secondary, icon, searchTerm, onClick, href } = props;
+    const classes = useStyles();
+    return (
+        <ListItem
+            alignItems="flex-start"
+            dense={true}
+            divider={true}
+            href={href}
+            onClick={onClick}
+            ref={ref}
+        >
+            <ListItemIcon className={classes.icon}>{icon}</ListItemIcon>
+            <ListItemText
+                primary={
+                    <Highlighter search={searchTerm}>{primary}</Highlighter>
+                }
+                primaryTypographyProps={{
+                    variant: "body1",
+                    color: "primary",
+                }}
+                secondary={
+                    <Highlighter search={searchTerm}>{secondary}</Highlighter>
+                }
+                secondaryTypographyProps={{
+                    variant: "caption",
+                    wrap: "true",
+                }}
+            />
+        </ListItem>
+    );
+});
+
+function DataSearchOption(props) {
+    const { selectedOption, searchTerm } = props;
+
+    const type = selectedOption._type;
+    let icon = <FolderIcon />;
+    let path = selectedOption._source.path;
+
+    //SS route to parent folder for the file util we have file viewers ready in sonora.
+    if (type === ResourceTypes.FILE) {
+        path = getParentPath(path);
+        icon = <DescriptionIcon />;
+    }
+
+    const href = `/${NavigationConstants.DATA}/${constants.DATA_STORE_STORAGE_ID}`;
+    const as = `/${NavigationConstants.DATA}/${constants.DATA_STORE_STORAGE_ID}${path}`;
+    return (
+        <Link href={href} as={as} passHref>
```

Comment:
more implementation details - https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-function-component